packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -256,6 +241,23 @@ const isLastBlockType = (editor, type) => {
   return false;
 };
 
+/**
+ * Function that finds the corresponding block key in the blocks object based on the targetType and targetLevel (if present) provided.
+ * @param {object} blocks - blocks object
+ * @param {string} type - type of the target block
+ * @param {number} level - level of the target block
+ * @returns string|undefined - block key or null if no matching block is found
+ */
+const getAnchorBlockKey = (blocks, { type, level = null }) => {
+  const anchorBlockKey = Object.keys(blocks).find((key) => {
+    const block = blocks[key];
+
+    return block.value.type === type && (level === null || block.value.level === level);
```

Comment:
Could you find a way to use matchNode from the block instead? It will avoid repeating the code we have there, and will be easier to manage when we add more types of blocks. I think generally we should avoid having logic specific to 1 type of block in the toolbar